### PR TITLE
gomodule multistage build

### PR DESCRIPTION
**Pull request by davidgamero**
# Description

update the gomodule dockerfile template to use a multistage build to reduce image size and improve security

### CODE REVIEW

File: template/dockerfiles/gomodule/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -v -o app-binary
 
-RUN go get
-RUN go build -v -o app
-RUN mv ./app /go/bin/
-
-CMD ["app"]
\ No newline at end of file
+FROM scratch
```
Comment (OliverMKing) [inline template/dockerfiles/gomodule/Dockerfile:11]: this is a breaking change. the golang base image includes debian bundled with some other tooling. Scratch does not.

many applications depend on having underlying OS features. Given Draft's purpose I'm not sure we should limit that.

I propose we use alpine or if we really want to go slim, distroless.

Comment (davidgamero) [reply template/dockerfiles/gomodule/Dockerfile:11]: alpine sounds good to keep the size small without removing all the tooling

**Review by OliverMKing — APPROVED** [review]

File: template/dockerfiles/gomodule/Dockerfile [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -v -o app-binary
 
-RUN go get
-RUN go build -v -o app
-RUN mv ./app /go/bin/
-
-CMD ["app"]
\ No newline at end of file
+FROM alpine
```
Comment (imiller31) [inline template/dockerfiles/gomodule/Dockerfile:11]: Any specific reason on why we use alpine? Should we use a distroless image?

**Review by bfoley13 — APPROVED** [review]

File: template/dockerfiles/gomodule/Dockerfile (hunk 2 above)
Comment (davidgamero) [reply template/dockerfiles/gomodule/Dockerfile:11]: we can do that I'll swap to something like `gcr.io/distroless/static-debian12` then it'll be even smaller

**Review by imiller31 — APPROVED** [review]

**Pull request merged**